src/charm.py

```diff
@@ -343,16 +342,20 @@ def configure_profile(self, profile_name):
                 with attempt:
                     self.k8s_resource_handler.lightkube_client.get(Namespace, name=profile_name)
         except RetryError:
-            self.log.error(f"Action failed. namespace {profile_name} was not created")
+            event.fail(f"Action failed. namespace {profile_name} was not created")
             return
+        try:
+            for file in PROFILE_CONFIG_FILES:
+                # TODO figure out which integrations are needed
+                yaml_text = self._safe_load_file_to_text(file)
+                self._apply_manifest(yaml_text, event, profile_name)
+        except LoadResourceError as e:
+            event.fail(
+                f"Failed to apply PodDefaults: CRD not defined. To fix, deploy admission webhook. Error: {e}"  # noqa E501
+            )
+        self._copy_seldon_secret(profile_name, event)
 
-        for file in PROFILE_CONFIG_FILES:
-            # TODO figure out which integrations are needed
-            yaml_text = self._safe_load_file_to_text(file)
-            self._apply_manifest(yaml_text, profile_name)
-        self._copy_seldon_secret(profile_name)
-
-    def _copy_seldon_secret(self, namespace):
+    def _copy_seldon_secret(self, namespace, event: ActionEvent):
         """Copy Seldon deployment secret from kubeflow namespace to the profile's namespace."""
         seldon_secret = None
         # check if seldon-core-mlflow integration secret exists in kubeflow namespace
```

Comment:
Do you know what creates the `name=mlflow-server-seldon-init-container-s3-credentials` that we're trying to copy here?  The name feels very long and specific, and also sounds like it comes from `mlflow` (which we *do not* include by default in the kubeflow bundle).  

I might be missing some context here.  It feels odd to include by default a configuration that tries to reuse something most kubeflow deployments wont have in the first place.

Reply:
it does come from mlflow-server charm. it's not in the bundle, that's why it's an optional configuration that would only be applied if mlflow is deployed.

Reply:
What @ca-scribner is raising here is a really valid concern that deserves a little bit more attention, though that change was introduced in another unrelated PR/commit, thus my suggestion would be to focus on what this PR is trying to fix (#92).

@NohaIhab - do you mind creating an issue to keep track of this concern?

Reply:
Yeah handling this as a separate issue sounds good to me, but @DnPlas I wonder if we should prioritise it before we actually release this feature.  Once it goes out in 1.7, changing this gets a bit messier due to upgrades

Reply:
As discussed, let's work on it and revisit the design of this method. @NohaIhab has filed some issues with our concerns about this design, and we should check them after 1.7. Thanks @ca-scribner